images/virtualization-artifact/pkg/controller/vi/internal/source/http.go

```diff
@@ -127,16 +139,119 @@ func (ds HTTPDataSource) Sync(ctx context.Context, vi *virtv2.VirtualImage) (boo
+
+		vi.Status.Phase = virtv2.ImageProvisioning
+		vi.Status.Progress = ds.statService.GetProgress(vi.GetUID(), pod, vi.Status.Progress)
+		vi.Status.Target.RegistryURL = ds.statService.GetDVCRImageName(pod)
+		vi.Status.DownloadSpeed = ds.statService.GetDownloadSpeed(vi.GetUID(), pod)
+
+		log.Info("Provisioning...", "progress", vi.Status.Progress, "pod.phase", pod.Status.Phase)
+	}
+
+	return true, nil
+}
+
+func (ds HTTPDataSource) StoreToPVC(ctx context.Context, vi *virtv2.VirtualImage) (bool, error) {
+	log, ctx := logger.GetDataSourceContext(ctx, "http")
+
+	condition, _ := service.GetCondition(vicondition.ReadyType, vi.Status.Conditions)
+	defer func() { service.SetCondition(condition, &vi.Status.Conditions) }()
+
+	supgen := supplements.NewGenerator(cc.VIShortName, vi.Name, vi.Namespace, vi.UID)
+	pod, err := ds.importerService.GetPod(ctx, supgen)
+	if err != nil {
+		return false, err
+	}
+	dv, err := ds.diskService.GetDataVolume(ctx, supgen)
+	if err != nil {
+		return false, err
+	}
+	pvc, err := ds.diskService.GetPersistentVolumeClaim(ctx, supgen)
+	if err != nil {
+		return false, err
+	}
+
+	switch {
+	case isDiskProvisioningFinished(condition):
+		log.Info("Image provisioning finished: clean up")
+
+		setPhaseConditionForFinishedImage(pvc, &condition, &vi.Status.Phase, supgen)
+
+		// Protect Ready Disk and underlying PVC.
+		err = ds.diskService.Protect(ctx, vi, nil, pvc)
+		if err != nil {
+			return false, err
+		}
+
+		// Unprotect import time supplements to delete them later.
+		err = ds.importerService.Unprotect(ctx, pod)
+		if err != nil {
+			return false, err
+		}
+
+		err = ds.diskService.Unprotect(ctx, dv)
+		if err != nil {
+			return false, err
+		}
+
+		return CleanUpSupplements(ctx, vi, ds)
+	case cc.AnyTerminating(pod, dv, pvc):
+		log.Info("Waiting for supplements to be terminated")
+	case pod == nil:
+		vi.Status.Progress = ds.statService.GetProgress(vi.GetUID(), pod, vi.Status.Progress)
+
+		envSettings := ds.getEnvSettings(vi, supgen)
+		err = ds.importerService.Start(ctx, envSettings, vi, supgen, datasource.NewCABundleForVMI(vi.Spec.DataSource))
+		var requeue bool
+		requeue, err = setPhaseConditionForImporterStart(&condition, &vi.Status.Phase, err)
+		if err != nil {
+			return false, err
+		}
+
+		log.Info("Create importer pod...", "progress", vi.Status.Progress, "pod.phase", "nil")
+
+		return requeue, nil
+	case !cc.IsPodComplete(pod):
+		log.Info("Provisioning to DVCR is in progress", "podPhase", pod.Status.Phase)
+
+		err = ds.statService.CheckPod(pod)
+		if err != nil {
+			return false, setPhaseConditionFromPodError(&condition, vi, err)
+		}
+
+		err = ds.importerService.Protect(ctx, pod)
+		if err != nil {
+			return false, err
+		}
+
+		vi.Status.Phase = virtv2.ImageProvisioning
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vicondition.Provisioning
+		condition.Message = "Import is in the process of provisioning to DVCR."
```

Comment:
DVCR -> PVC

Reply:
It's ok

Reply:
No, this code block is responsible for uploading to DVCR